### ISSUE TITLE
Using dynamic=True in a keras Metric constructor should run the metric methods in eager mode

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [tensorflow/tensorflow#39114]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.2.0rc4
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**

Currently, when using `dynamic=True` in a Metric constructor, it has no effect. All the methods are run in graph mode. Since this feature is [already present in keras Layers](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#arguments_20), and that a Metric is a subclass of Layer, one would expect a metric to run in eager mode if `dynamic=True` is provided.

**Will this change the current api? How?**

No change to the API. dynamic=True is already accepted in a Metric constructor. The change would be in behavior only. The flag should trigger the eager mode. Currently it doesn't do anything.

**Who will benefit with this feature?**

Users who want to make custom metrics which need eager mode and want to distribute them.

**Any Other info.**

Here is a minimal notebook reproducing the problem. With this feature implemented, the error should not happen:

https://colab.research.google.com/drive/11EuSPoyedceHySz_uxxqjZJbU2uatpfk

**Comment by jpodivin** [comment]
Intuitively it might be easiest to just pass dynamic arg to constructor. Assuming that the Metric class inherits all relevant infrastructure from the Layer class. But looking at the Layer class code it seems that approach wouldn't work.

https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/engine/base_layer.py#L357-L360

Looking at the other metrics that inherit from Metric, it seems they use metrics_utils to handle to handle the heavy lifting. 
For example in the case of the Recall class:

https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/metrics.py#L1382-L1392
So that could be the path forward.

**Comment by jvishnuvardhan** [comment]
@gabrieldemarmiesse This feature not yet implemented. Is there any actionable item like raising PR? 

Please note that Keras development moved to another repository to focus entirely on only keras. Could you please repost this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues). Thanks!

**Comment by gabrieldemarmiesse** [comment]
I supposed the path forward would be to make a pull request. It's been quite a while since I last worked with deep learning so I'll let users who are interested implement it.

**Comment by jvishnuvardhan** [comment]
I opened this issue in keras-team/keras. Let's track the progress there https://github.com/keras-team/keras/issues/15758
